### PR TITLE
El update master

### PR DESCRIPTION
**Pull request by RealNobody**
I was curious if we were out of date with the source.

This is the merge with the original source.  Don't know if we want to merge it or not, but I figured I'd put the PR out there and see what people say.

### CODE REVIEW

**Comment by subramantr** [comment]
I am still trying to understand configuration module better.  All I know is, this(`Configuration`) module is being used in `lib/config/application.rb`(properties are being overridden in partner specific and env. specific application.rb(s)).
Also, I am not really sure what this PR is doing.

It would be nice to know (at least in layman's terms) on what are we trying to achieve here and esp. how is this going to be benificial.

Thanks

**Comment by donovanbray** [comment]
It looks like luke put in a single commit that made our configuration gem behave with 1.9.3.  This probably should have been PR'd BACK to the upstream gem for inclusion. I don't know if @LukeHammond  did.

but otherwise it looks like @RealNobody is just bringing our fork up to date with the source.

What our fork has that the original doesn't:

https://github.com/ahoward/configuration/compare/demandchain:master...ahoward:master

What ahoward's original has that ours doesn't.

https://github.com/demandchain/configuration/compare/ahoward:master...demandchain:master

which looks exactly like what @RealNobody is attempting to add to ours

so I'm :+1:  but also strongly suggest @LukeHammond  PR's his changes to ahoward if he hasn't.  We will need to maintain our own fork until ahoward fixes his one way or the other.

**Comment by RealNobody** [comment]
> also strongly suggest @LukeHammond PR's his changes to ahoward

I strongly disagree.

Our change says to take the return of a function which is an array of symbols (which is generally preferred over strings), and to convert them to strings.

If I were ahoward, I would reject this PR outright.

Personally, we should just fix our code like we should have done in the first place.  I find it amazing that we actually specifically require strings over symbols, or are too lazy to convert them to strings when we need to.

It's our problem, and we should have fixed it correctly in the first place.

This PR was just to try to keep us up to date with ahoward as the easiest way of continuing to not deal with the real problem.
